src/contracts/services/lend.ts

```diff
@@ -0,0 +1,512 @@
+    silkMaxAllowance: convertCoinFromUDenom(allowance.max, NormalizationFactor.LEND).toString(),
+    silkAllowanceUsed: convertCoinFromUDenom(allowance.used, NormalizationFactor.LEND).toString(),
+    maxLtv: Number(maxLtv),
+    // Collateral is expressed differently depending on vault type
+    collateral: {
+      total:
+    (vaultVersion === VaultVersion.V1 || vaultVersion === VaultVersion.V2)
+      ? convertCoinFromUDenom(elasticCollateral, NormalizationFactor.LEND).toString()
+      : convertCoinFromUDenom(
+        BigNumber(elasticCollateral).plus(safeCollateral),
+        NormalizationFactor.LEND,
+      ).toString(),
+      elastic: convertCoinFromUDenom(elasticCollateral, NormalizationFactor.LEND).toString(),
+      base: convertCoinFromUDenom(baseCollateral, NormalizationFactor.LEND).toString(),
+      safe: convertCoinFromUDenom(safeCollateral, NormalizationFactor.LEND).toString(),
+      lastAccruedAt: new Date(Number(collateralLastAccrued) * 1000),
+      oracleDelay: Number(collateralOracleDelay),
+    },
+    debt: {
+      total: convertCoinFromUDenom(debt.elastic, NormalizationFactor.LEND).toString(),
+      base: convertCoinFromUDenom(debt.base, NormalizationFactor.LEND).toString(),
+      lastAccruedAt: new Date(Number(debt.last_accrued) * 1000),
+    },
+    interestRate: {
+      current: Number(interestRate.current),
+      target: Number(interestRate.target),
+      delta: Number(interestRate.delta),
+      ratePerSecond: Number(interestRate.rate_per_second),
+      lastUpdatedAt: new Date(
+        Number(interestRate.last_changed) * 1000,
+      ),
+    },
+    borrowFee: {
+      current: Number(borrowFee.current),
+      target: Number(borrowFee.target),
+      delta: Number(borrowFee.delta),
+      ratePerSecond: Number(borrowFee.rate_per_second),
+      lastUpdatedAt: new Date(Number(borrowFee.last_changed) * 1000),
+    },
+    liquidationFee: {
+      discount: Number(liquidationFee.discount),
+      minimumDebt: convertCoinFromUDenom(
+        liquidationFee.min_debt,
+        NormalizationFactor.LEND,
+      ).toString(),
+      daoShare: Number(liquidationFee.treasury_share),
+      callerShare: Number(liquidationFee.caller_share),
+    },
+    isProtocolOnly: isProtocol,
+    status: vault.status,
+    openPositions: Number(openPositions.value),
+    totalPositions: Number(positionIdCounter.value),
+    whitelist: vault.whitelist,
+  } as Vault;
+}
+
+/**
+* Parse lend vaults response
+*/
+function parseLendVaults(vaults: VaultsResponse, vaultVersion: VaultVersion) {
+  return vaults.vaults.reduce((prev, vault) => {
+    const {
+      id: vaultId,
+    } = vault.vault;
+    return {
+      ...prev,
+      [vaultId]: parseLendVault(vault, vaultVersion),
+    };
+  }, {} as Vaults);
+}
+
+/**
+ * parses the vaults reponse from a batch query of
+ * multiple vaults contracts
+ */
+const parseBatchQueryVaultsInfo = (
+  response: BatchQueryParsedResponse,
+  vaultVersions: VaultVersion[],
```

Comment:
I really dislike that vaultVersions is a required argument.... we really should find a way to make it work without it

Reply:
also dislike it, but we have no choice and need to be clear about the differences between the different vaults.